packages/lib/services/searchengine/SearchEngineUtils.ts

```diff
@@ -48,11 +49,32 @@ export default class SearchEngineUtils {
 		// handle it here by checking if `user_updated_time` IS NOT NULL. Was causing this
 		// issue: https://discourse.joplinapp.org/t/how-to-recover-corrupted-database/9367
 		if (noteIds.length !== notes.length) {
-			// remove null objects
-			return sortedNotes.filter(n => n);
+			const results1: any = [];
+			const filtered_notes = sortedNotes.filter(n => n);
+			if (!Setting.value('showCompletedTodos')) {
```

Comment:
It looks like this block here, and the block starting on line 67 are identical. Can you come up with a way to only do this filtering once? You should also look into using the  [`filter`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter) function instead of a `for` loop.